### PR TITLE
Port missing external API validation

### PR DESCRIPTION
**Pull request by colinking**
## Description

I reviewed the external API endpoints and validated their validation logic aligns with what is in our production API. Thankfully there wasn't much missing -- the only non-trivial change is validating that workflows do not exceed the maximum child run limit.

For tasks execute and prompt creation/submission, we do not perform validate parameter schemas nor values because that logic lives in airport. In a follow-up PR, I'll move that logic into the CLI and then update both endpoints to use it.

## Test plan

Ran a task locally and confirmed it was able to execute other tasks.

### CODE REVIEW

**Comment by colinking** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1071** <a href="https://app.graphite.dev/github/pr/airplanedev/cli/1071" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airplanedev/cli/1071?utm_source=stack-comment).

File: pkg/server/apiext/tasks.go [hunk 1]
```diff
@@ -0,0 +1,357 @@
+package apiext
```
Comment (colinking) [inline pkg/server/apiext/tasks.go:1]: I split out the task handlers into their own file.

File: pkg/server/apiext/tasks.go [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+		// Pull env slug from the parent run.
+		parentRun, err := state.GetRunInternal(ctx, parentID)
+		if err != nil {
+			return api.RunTaskResponse{}, err
+		}
+		if parentRun.FallbackEnvSlug != "" {
+			envSlug = &parentRun.FallbackEnvSlug
+		}
+
+		// Check if this run has exceeded the max child runs limit.
```
Comment (colinking) [inline pkg/server/apiext/tasks.go:62]: Added this validation logic. cc @kellymtrinh

File: pkg/server/apiext/tasks.go [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+				return api.RunTaskResponse{}, libhttp.NewErrBadRequest("Parent run has exceeded the maximum limit of %d child runs", maxWorkflowChildRuns)
+			}
+		}
+	} else {
+		envSlug = serverutils.GetEffectiveEnvSlugFromRequest(state, r)
+	}
+
+	localTaskConfig, ok := state.TaskConfigs.Get(req.Slug)
+	isBuiltin := builtins.IsBuiltinTaskSlug(req.Slug)
+	if !isBuiltin && !ok {
```
Comment (colinking) [inline pkg/server/apiext/tasks.go:78]: I inverted this if statement to make it easier to read. It's a no-op though.

File: pkg/server/apiext/tasks.go [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+
+// GetTaskMetadataHandler handles requests to the /v0/tasks/getMetadata endpoint. It generates a deterministic task ID
+// for each task found locally, and its primary purpose is to ensure that the task discoverer does not error.
+// If a task is not local, it tries the fallback environment, so that local views
+// can route correctly to the correct URL.
+func GetTaskMetadataHandler(ctx context.Context, state *state.State, r *http.Request) (libapi.TaskMetadata, error) {
+	slug := r.URL.Query().Get("slug")
+	if slug == "" {
+		return libapi.TaskMetadata{}, libhttp.NewErrBadRequest("expected a slug")
+	}
```
Comment (colinking) [inline pkg/server/apiext/tasks.go:299]: added this validation logic

File: pkg/server/apiext/tasks.go [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+		Slug:    slug,
+		IsLocal: true,
+	}, nil
+}
+
+func GetTaskReviewersHandler(ctx context.Context, state *state.State, r *http.Request) (api.GetTaskReviewersResponse, error) {
+	taskSlug := r.URL.Query().Get("taskSlug")
+	if taskSlug == "" {
+		return api.GetTaskReviewersResponse{}, libhttp.NewErrBadRequest("expected a slug")
+	}
```
Comment (colinking) [inline pkg/server/apiext/tasks.go:326]: added this validation logic

File: pkg/server/apiext/tasks.go [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+		if err != nil {
+			return api.GetTaskReviewersResponse{}, err
+		}
+		return api.GetTaskReviewersResponse{
+			Task: &libapi.Task{
+				Slug:       taskSlug,
+				Parameters: parameters,
+				Triggers:   []libapi.Trigger{{Kind: "form"}},
+			},
+			Reviewers: []api.ReviewerID{},
```
Comment (colinking) [inline pkg/server/apiext/tasks.go:340]: I initialized this so it's not null

File: pkg/server/apiext/apiext.go [hunk 7]
```diff
@@ -452,15 +139,13 @@ type CreateDisplayResponse struct {
...
-		return CreateDisplayResponse{}, libhttp.NewErrBadRequest("expected a X-Airplane-Token header")
-	}
-	claims, err := dev.ParseInsecureAirplaneToken(token)
+	runID, err := getRunIDFromToken(r)
 	if err != nil {
-		return CreateDisplayResponse{}, libhttp.NewErrBadRequest("invalid airplane token: %s", err.Error())
+		return CreateDisplayResponse{}, err
+	}
+	if runID == "" {
+		return CreateDisplayResponse{}, libhttp.NewErrBadRequest("this endpoint can only be called from the task runtime")
```
Comment (calufornia) [inline pkg/server/apiext/apiext.go:147]: How come the error message changed here? `runID` being empty => it's not being called from the task runtime?

File: pkg/server/apiext/tasks.go (hunk 6 above)
Comment (calufornia) [reply pkg/server/apiext/tasks.go:340]: Why do we need to do this?

**Review by calufornia — APPROVED** [review]

File: pkg/server/apiext/tasks.go (hunk 6 above)
Comment (colinking) [reply pkg/server/apiext/tasks.go:340]: This matches how it works in the production API.

File: pkg/server/apiext/apiext.go (hunk 7 above)
Comment (colinking) [reply pkg/server/apiext/apiext.go:147]: To match what we return in the production API. Also this should not have allowed the `runID` claim to be empty.

**Pull request merged**